### PR TITLE
added files to programming_4

### PR DESCRIPTION
**Pull request by MariaPonomarenko38**
Варіант 20: Клас ОНЛАЙН ЗУСТРІЧ: id, date, start_time, end_time, meeting_url, owner (first and last name), participant (first and last name).

### CODE REVIEW

File: programming/programming_4/main.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
+from Ponomarenko_pmi25.programming.programming_4.meetings import Meetings, fill_list_from_file
+from Ponomarenko_pmi25.programming.programming_4.validation import validate_input
+
+
+def edit_file(file, id, val):
```
Comment (tetianastoyko) [inline programming/programming_4/main.py]: move to class

File: programming/programming_4/main.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    arg = val.split()
+    arg[0] = int(arg[0])
+    f = open(file, "r+")
```
Comment (tetianastoyko) [inline programming/programming_4/main.py]: check whether file is exist

File: programming/programming_4/meetings.py [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+                elif param in ['participant', 'owner'] and x.get_name().lower() > y.get_name().lower():
+                    self.meetings[j], self.meetings[j + 1] = self.meetings[j + 1], self.meetings[j]
+                elif param == 'date':
+                    a = dt.strptime(x, "%d.%m.%Y")
+                    b = dt.strptime(y, "%d.%m.%Y")
+                    if a > b:
+                        self.meetings[j], self.meetings[j + 1] = self.meetings[j + 1], self.meetings[j]
+
+    def add(self, file, val):
+        with open(file, 'a', encoding='utf-8') as f:
```
Comment (tetianastoyko) [inline programming/programming_4/meetings.py]: check whether file is exist

File: programming/programming_4/meetings.py [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+            f.write("\n" + val)
+        arg = val.split()
+        arg[0] = int(arg[0])
+        self.add_to_list(OnlineMeeting(*arg))
+
+    def remove(self, file, id):
+        f = open(file, "r+")
```
Comment (tetianastoyko) [inline programming/programming_4/meetings.py]: check whether file is exist

File: programming/programming_4/online_meeting.py [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+        self.end_time = end_time
+        self.meeting_url = meeting_url
+        self.owner = FullName(owner_name, owner_surname)
+        self.participant = FullName(participant_name, participant_surname)
+
+    def __repr__(self):
+        fields = ['id', 'date', 'start_time', 'end_time', 'meeting_url', 'owner', 'participant']
+        answer = [str(self.__id), self.date, self.start_time, self.end_time, self.meeting_url, self.owner, self.participant]
+        s = ''
+        for i in range(len(answer)):
```
Comment (tetianastoyko) [inline programming/programming_4/online_meeting.py]: good solution

File: programming/programming_4/meetings.py [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+
+        f.close()
+        if len(self.meetings) > 1:
+            for i in range(len(self.meetings)):
+                if self.meetings[i].get_field('id') == int(id):
+                    self.meetings[i] = OnlineMeeting(*arg)
+                    return
+
+
+def fill_list_from_file(file, param):
```
Comment (tetianastoyko) [inline programming/programming_4/meetings.py]: move to class

**Comment by MariaPonomarenko38** [comment]
Я не перевіряла чи є файл в кожному методі який пов'язаний з файлом, а зробила це один раз на початку роботи програми

File: programming/programming_4/meetings.py (hunk 3 above)
Comment (MariaPonomarenko38) [reply programming/programming_4/meetings.py]: I decided not to check in every method if file exists , but did it only once when user inputs name of the file

File: programming/programming_4/meetings.py [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+        for i in self.meetings:
+            if i.search(value):
+                print(i)
+
+    def sort(self, param):
+        for i in range(len(self.meetings)):
+            for j in range(len(self.meetings) - i - 1):
+                x = self.meetings[j].get_field(param)
+                y = self.meetings[j + 1].get_field(param)
+                if param not in ['participant', 'owner', 'date'] and x > y:
```
Comment (tetianastoyko) [inline programming/programming_4/meetings.py]: I have already seen it in Malashniak code

File: programming/programming_4/online_meeting.py [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+            if answer[i] is not None:
+                s += fields[i] + ': ' + str(answer[i]) + '\n'
+            else:
+                s += fields[i] + ': ' + 'none' + '\n'
+        return s
+
+    def compare(self, other):
+        for param in {'__id', 'date', 'start_time', 'end_time', 'meeting_url', 'owner', 'participant'}:
```
Comment (tetianastoyko) [inline programming/programming_4/online_meeting.py]: create dictionary in 1 place and use it.
this line 
53
59
36
.....

File: programming/programming_4/validation.py [hunk 9]
```diff
@@ -0,0 +1,171 @@
...
+        self.validate_id()
+        if s != 'edit':
+            self.validate_unique()
+        self.validate_date()
+        self.validate_time()
+        self.validate_url()
+        self.validate_name()
+
+    def validate_file_while_filling(self):
+        try:
```
Comment (tetianastoyko) [inline programming/programming_4/validation.py]: lines of this method and validate_input are copy pasted

**Pull request merged**